### PR TITLE
cppclient: batch calls to getTrytes/findTransactions

### PR DESCRIPTION
**Pull request by tsvisabo**

### CODE REVIEW

File: cppclient/api_json.cc [hunk 1]
```diff
@@ -36,6 +36,14 @@ using iota::model::TryteTransaction;
 DEFINE_uint32(maxNumAddressesForGetBalances, 1000,
               "Maximum number of addresses to query for in 'getBalances'");
 
+DEFINE_uint32(maxQuerySizeFindTransactions, 500,
+              "Maximum number of addresses/bundles/approvees to query for in "
+              "'findTransactions'");
+
+DEFINE_uint32(maxQuerySizeGetTrytes, 100,
```
Comment (th0br0) [inline cppclient/api_json.cc]: this is `1000` on IRI by default

File: cppclient/api_json.cc [hunk 2]
```diff
@@ -36,6 +36,14 @@ using iota::model::TryteTransaction;
 DEFINE_uint32(maxNumAddressesForGetBalances, 1000,
               "Maximum number of addresses to query for in 'getBalances'");
 
+DEFINE_uint32(maxQuerySizeFindTransactions, 500,
```
Comment (th0br0) [inline cppclient/api_json.cc:39]: this can probably be increased to 5000 or so as well

File: cppclient/tests/beast.cc [hunk 3]
```diff
@@ -39,11 +39,11 @@ TEST_F(BeastIotaAPITest, InvalidHostPort) {
 
 TEST_F(BeastIotaAPITest,
        TestgetConfirmedBundlesForAddressesOnInvalidAddressNotCrashing) {
-  MockBeast beast("localhost", 15265);
+  MockBeast beast("iri01.mainnet.iota.cafe", 14265);
```
Comment (jserv) [inline cppclient/tests/beast.cc]: Why hardcode host name here?

Comment (tsvisabo) [reply cppclient/tests/beast.cc]: you got a point

Comment (tsvisabo) [reply cppclient/tests/beast.cc]: you got a point ;)

File: cppclient/api_json.cc [hunk 4]
```diff
@@ -133,44 +141,87 @@ std::vector<std::string> IotaJsonAPI::findTransactions(
...
+      LOG(INFO) << __FUNCTION__ << " request failed.";
+      return {};
+    }
 
-  if (!maybeResponse.value()["hashes"].is_null()) {
-    return maybeResponse.value()["hashes"].get<std::vector<std::string>>();
+    if (!maybeResponse.value()["hashes"].is_null()) {
+      auto resVec =
+          maybeResponse.value()["hashes"].get<std::vector<std::string>>();
+      txs.insert(std::end(txs), std::begin(resVec), std::end(resVec));
```
Comment (th0br0) [inline cppclient/api_json.cc]: you should use `std::copy` here?

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: cppclient/api_json.cc (hunk 2 above)
Comment (tsvisabo) [reply cppclient/api_json.cc:39]: it failed on 2080

Comment (th0br0) [reply cppclient/api_json.cc:39]: fair enough, let's keep it this low then.

**Review by th0br0 — APPROVED** [review]

**Pull request merged**